### ISSUE TITLE
Memory Leak due to readFragment

### ISSUE THREAD

**Issue by ryansydnor** [apollographql/apollo-client#3132]
**Intended outcome:**
When using `readFragment` against data that is already in the cache, node's memory usage should remain stable.

**Actual outcome:**
Our node server's memory usage spiked once we started using `readFragment` in a heavily trafficked part of our application:

![image](https://user-images.githubusercontent.com/1832096/37190324-24a4e7e8-2327-11e8-9193-b0bf8d53934c.png)

**How to reproduce the issue:**
```

const ApolloClient = require('apollo-client').ApolloClient
const HttpLink = require('apollo-link-http').HttpLink;
const InMemoryCache = require('apollo-cache-inmemory').InMemoryCache;
const gql = require('graphql-tag');
const fetch = require('node-fetch');

const client = new ApolloClient({
  link: new HttpLink({
    uri: 'https://www.teacherspayteachers.com/graph/graphql',
    fetch
  }),
  cache: new InMemoryCache()
});

client.query({
  query: gql`
    query ProductTest {
      products(ids:[968647]) {
        id
        description
        name
      }
    }
  `,
}).then(data => leakMemory());

const leakMemory = () => {
  setInterval(() => {
      client.readFragment({
        id: 'Product:968647',
        fragment: gql`
          fragment Leak on Product {
            id
            name
          }
        `
      });
      console.log(process.memoryUsage());
  }, 5);
};
```

I've tracked the issue down to this line: https://github.com/apollographql/apollo-client/blob/311d2057ad8455795f1d42374312ad4d95e83d1e/packages/apollo-utilities/src/transform.ts#L184 

We are never getting cache hits despite using exactly the same fragment because I believe we are losing our equality reference due to object instantiation here: https://github.com/apollographql/apollo-client/blob/311d2057ad8455795f1d42374312ad4d95e83d1e/packages/apollo-utilities/src/fragments.ts#L68.

**Version**
- apollo-client@2.2.6

/label memory-leak

**Comment by jonmagic** [comment]
We're experiencing the same bug I think, I noticed while memory profiling:

<img width="862" alt="cloudapp annotation 2018-03-30 at 10 20 06 am png" src="https://user-images.githubusercontent.com/623/38157189-5e8e90ee-3439-11e8-8e04-fc90e8cf67c8.png">

**Comment by edqd** [comment]
Is it possible that what we're experiencing is a very similar/same bug, but without any usage of readFragment? Although we do use cacheRedirects.
#### Memory usage ####
![screen shot 2018-04-18 at 11 27 09](https://user-images.githubusercontent.com/4508655/38923751-928623c8-42fb-11e8-9703-146a352d9d41.png)

#### Memory profiling ####
![screen shot 2018-04-18 at 11 23 01](https://user-images.githubusercontent.com/4508655/38923593-4105508c-42fb-11e8-81fd-dbe40f7063b4.png)

**Comment by jakepusateri** [comment]
I think https://github.com/apollographql/apollo-client/pull/3444 fixed this issue by removing the internal caches in this file. It should be in the next release of apollo-utilities, having been merged to master already.

**Comment by hwillson** [comment]
Thanks for reminding us about this issue @jakepusateri. Yes, this should now be fixed and will be included in the release of `apollo-client` 2.3.2 this Tuesday (May 29th, 2018).
